verification/src/types/mod.rs

```diff
@@ -192,8 +192,9 @@ impl core::Client {
         }
 
         let headers_mmr_root = packed_proof_update.new_headers_mmr_root().unpack();
+        let id = prev_client_opt.map(|client| client.id).unwrap_or(0);
         let new_client = Self {
-            id: 0,
+            id,
```

Comment:
I think it's the correct id from the perspective of this function. However, we still need to set the expected client id in the contract because we are actually replacing the oldest client.

Reply:
> When do update, the id of output client must be same as the input client.
I find it very tricky that the input client is the oldest client and we are obtaining new-latest client with old-latest client.
I think the expected output client is a client with new-latest content and oldest client's id.
If we don't set the id of the updated client in the update contract, it will have the id of the old-latest client, which is not the same as the input client.